web-platform/website/src/services/AuthenticationService.tsx

```diff
@@ -73,25 +72,13 @@ export async function getCurrentUser() {
 
 export function AuthenticationProvider(props: { children: React.ReactNode }) {
   const [user, setUser] = useState<User | undefined>();
-  const [isUserVerified, setIsUserVerified] = useState(false);
   const [isLoggingIn] = useState(false);
   const [isLoaded, setIsLoaded] = useState(false);
 
   const { auth, analytics } = useFirebase();
 
   useEffect(() => {
     return onAuthStateChanged(auth, (firebaseUser) => {
-      setIsUserVerified(
-        Boolean(
-          firebaseUser &&
-            firebaseUser.providerData &&
-            firebaseUser.providerData.length &&
-            ((firebaseUser.providerData[0].providerId === "password" &&
-              firebaseUser.emailVerified) ||
-              firebaseUser.providerData[0].providerId === "google.com")
-        )
-      );
-
```

Comment:
We can restore this file and then do just this:
```
onAuthStateChanged((auth, firebaseUser) => {
    // Refactor code in updateUserVerificationState
    updateUserVerificationState(firebaseUser);
});

useEffect(() => {
  (async () => {
       updateUserVerificationState(user && user.firebaseUser);
   })();
}, []); 
```

Reply:
I considered this, but I think it would be a bad pattern to have to always manually call `updateUserVerificationState` before looking at the value of `isUserVerified`.

Reply:
implemented the suggested change, and exported a `reloadUser` to manually trigger